test/fetcher_test.exs

```diff
@@ -11,6 +11,30 @@ defmodule MozartFetcher.FetcherTest do
 
   test "it gets success when passing component config" do
     assert Fetcher.process([%Config{endpoint: "http://localhost:8082/foo", id: "foo"}]) ==
-             "{\"components\":[{\"envelope\":{\"bodyInline\":\"\",\"bodyLast\":[],\"head\":[]},\"id\":\"foo\",\"index\":0,\"status\":200}]}"
+      ~s({"components":[{"envelope":{"bodyInline":"","bodyLast":[],"head":[]},"id":"foo","index":0,"status":200}]})
+  end
+
+  test "it returns envelopes for components timeout" do
+    config = [
+      %Config{endpoint: "http://localhost:8082/foo", id: "foo"},
+      %Config{endpoint: "http://localhost:8082/timeout", id: "timeout"}
+    ]
+
+    expected =
+      ~s({"components":[{"envelope":{"bodyInline":"","bodyLast":[],"head":[]},"id":"foo","index":0,"status":200},{"envelope":{"bodyInline":"","bodyLast":[],"head":[]},"id":"timeout","index":1,"status":408}]})
+
+    assert Fetcher.process(config) == expected
+  end
+
+  test "it returns empty envelopes for components erroring out with no generated response" do
+    config = [
+      %Config{endpoint: "http://localhost:8082/foo", id: "foo"},
+      %Config{endpoint: "http://localhost:8082/timeout", id: "timeout"}
+    ]
+
+    expected =
+      ~s({"components":[{"envelope":{"bodyInline":"","bodyLast":[],"head":[]},"id":"foo","index":0,"status":200},{"envelope":{"bodyInline":"","bodyLast":[],"head":[]},"id":"timeout","index":null,"status":null}]})
```

Comment:
Why do we lose the index for the empty ones?

Reply:
the order is guaranteed but we don't have the `index` value in the configs. I don't think it's necessary. Otherwise, it requires an additional Enumeration.